app/scripts/components/dashboard/charts/stack-bar-chart.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { PropTypes } from 'react';
+import {BarChart} from 'react-d3';
+import * as func from '../../../utils/functional';
+
+require('stylesheets/dashboard/charts/stack-bar-chart');
+
+const StackBarChart = React.createClass({
+  propTypes: {
+    data: PropTypes.object,
+  },
+  render() {
+    const dataRes = func.Result.aggregate(this.props.data, 'STATUS', 'REGION');
+    return dataRes.andThen(data => {
+      if (data.keys.aggProps.length > 0) {
+        const barData = Object.keys(data)
+          .filter(key => key !== 'keys')
+          .map(key => {
+            return {
+              name: key,
+              values: data.keys.sumProps.map(status => {
+                return {
+                  x: status,
+                  y: data[key][status] ? data[key][status] : 0,
+                };
+              }),
+            };
+          });
```

Comment:
So, continuing from sumAggregate comments below, if we have `data` as
```js
Ok({
  keys: {
    aggProps: ['x', 'y', 'z'],
    sumProps: ['m', 'n', 'o'],
  },
  x: {m: 1, n: 1},
  y: {n: 1},
  z: {o: 2}
});
```

this filter, map makes it into

```js
[
  {name: 'x', values: [{x: 'm', y: 1}, {x: 'n', y: '1'}]},
  {name: 'y', values: [{x: 'n', y: 1}]},
  {name: 'z', values: [{x: 'o', y: 2}]},
]
```

I think?